src/SimpleNotePad.java

```diff
@@ -0,0 +1,181 @@
+	private JMenuItem newFile;
+	private JMenuItem saveFile;
+	private JMenuItem printFile;
+	private JMenuItem openFile;
+	private JMenuItem copy;
+	private JMenuItem paste;
+	private JMenuItem replace;
+	private JMenuItem[] recent;
+	private Integer count;
+
+	// constructor
+	public SimpleNotePad() {
+		instantiateMenuObjects(); // instantiates the menu objects
+		addMenuItems(); // adds menu items to menubar
+		configureMenuBars();
+		configureActionListeners();
+		configureJFrameSettings();
+	}
+
+	// instantiates the menu objects
+	private void instantiateMenuObjects() {
+		menuBar = new JMenuBar();
+		fileMenu = new JMenu("File");
+		editMenu = new JMenu("Edit");
+		recentFile = new JMenu("Recent");
+		notePad = new JTextPane();
+		newFile = new JMenuItem("New File");
+		saveFile = new JMenuItem("Save File");
+		printFile = new JMenuItem("Print File");
+		openFile = new JMenuItem("Open File");
+		copy = new JMenuItem("Copy");
+		paste = new JMenuItem("Paste");
+		replace = new JMenuItem("Replace");
+		recent = new JMenuItem[3];
+
+		count = 0;
+	}
+
+	// configures the JFrame settings
+	private void configureJFrameSettings() {
+		setTitle("A Simple Notepad Tool");
+		add(new JScrollPane(notePad));
+		setPreferredSize(new Dimension(600, 600));
+		setDefaultCloseOperation(JFrame.EXIT_ON_CLOSE);
+		setVisible(true);
+		pack();
+
+	}
+
+	// adds actions listeners
+	private void configureActionListeners() {
+		setActionListeners(newFile, "new");
+		setActionListeners(saveFile, "save");
+		setActionListeners(printFile, "print");
+		setActionListeners(openFile, "open");
+		setActionListeners(copy, "copy");
+		setActionListeners(paste, "paste");
+		setActionListeners(replace, "replace");
+
+	}
+
+	// Sets the action listener and commands based on item and command text
+	private void setActionListeners(JMenuItem item, String command) {
+
+		item.addActionListener(this);
+		item.setActionCommand(command);
+	}
+
+	// adds menu items
+	private void addMenuItems() {
+
+		addFileMenuItems();
+		addEditMenuItems();
+
+	}
+
+	// adds file menu items
+	private void addFileMenuItems() {
+		fileMenu.add(newFile);
+		fileMenu.addSeparator();
+		fileMenu.add(saveFile);
+		fileMenu.addSeparator();
+		fileMenu.add(printFile);
+		fileMenu.addSeparator();
+		fileMenu.add(openFile);
+		fileMenu.addSeparator();
+		fileMenu.add(recentFile);
+
+	}
+
+	// adds edit menu items
+	private void addEditMenuItems() {
+
+		editMenu.add(copy);
+		editMenu.addSeparator();
+		editMenu.add(paste);
+		editMenu.addSeparator();
+		editMenu.add(replace);
+
+	}
+
+	// configure the menubars by adding menus
+	private void configureMenuBars() {
+		menuBar.add(fileMenu);
+		menuBar.add(editMenu);
+		setJMenuBar(menuBar);
+	}
+
+	@Override
+	// Handles the Action Events based on action event keyword
+	public void actionPerformed(ActionEvent e) {
+
+		// Checks what the action event command is and creates a specific Action Object
+		// to execute the action
```

Comment:
The overall design of this program is very similar to the command design pattern approach. You made an abstract class called 'Action', which is the command interface. The subclasses of Action implements the doAction method.  It appears that most of the doAction methods takes in a JTextPane as a parameter. If possible, we should aim to see  we can make the doAction method take in JTextPane as the only parameter for all Action subclasses. If we can accomplish this, we can make an abstraction and significantly increase the readability and extensibility. 

So instead of using a series of if and else statements, we can just write something like...
```suggestion
		// to execute the action
		Action action = (Action)e.getSource();
		action.doAction(notePad);
```


The link below provides great guidance on how to incorporate command design pattern. We should become more familiar with common design pattern in order to grow as software developers.
[https://alvinalexander.com/java/java-command-design-pattern-in-java-examples](url)